utils/results/common.go

```diff
@@ -0,0 +1,612 @@
+	for _, cveObj := range cves {
+		cveRows = append(cveRows, formats.CveRow{Id: cveObj.Id, CvssV2: cveObj.CvssV2Score, CvssV3: cveObj.CvssV3Score})
+	}
+	return cveRows
+}
+
+// FindMaxCVEScore returns the maximum CVSS score of the given CVEs or score based on severity and applicability status if not exists.
+func FindMaxCVEScore(severity severityutils.Severity, applicabilityStatus jasutils.ApplicabilityStatus, cves []formats.CveRow) (string, error) {
+	if len(cves) == 0 {
+		return fmt.Sprintf("%.1f", severityutils.GetSeverityScore(severity, applicabilityStatus)), nil
+	}
+	maxCve := severityutils.MinCveScore
+	for _, cve := range cves {
+		cveScore, err := GetCveScore(severity, applicabilityStatus, cve)
+		if err != nil {
+			return "", err
+		}
+		if cveScore > maxCve {
+			maxCve = cveScore
+		}
+		// if found maximum possible cve score, no need to keep iterating
+		if maxCve == severityutils.MaxCveScore {
+			break
+		}
+	}
+	return fmt.Sprintf("%.1f", maxCve), nil
+}
+
+// GetCveScore returns the CVSS score of the given CVE or score based on severity and applicability status if not exists.
+func GetCveScore(severity severityutils.Severity, applicabilityStatus jasutils.ApplicabilityStatus, cve formats.CveRow) (float32, error) {
+	if cve.CvssV3 == "" {
+		return severityutils.GetSeverityScore(severity, applicabilityStatus), nil
+	}
+	score, err := strconv.ParseFloat(cve.CvssV3, 32)
+	return float32(score), err
+}
+
+func GetViolatedLicenses(allowedLicenses []string, licenses []services.License) (violatedLicenses []services.Violation) {
+	if len(allowedLicenses) == 0 {
+		return
+	}
+	for _, license := range licenses {
+		if !slices.Contains(allowedLicenses, license.Key) {
+			violatedLicenses = append(violatedLicenses, services.Violation{
+				LicenseKey:    license.Key,
+				LicenseName:   license.Name,
+				Severity:      severityutils.Medium.String(),
+				Components:    license.Components,
+				IssueId:       customLicenseViolationId,
+				WatchName:     fmt.Sprintf("jfrog_%s", customLicenseViolationId),
+				ViolationType: formats.ViolationTypeLicense.String(),
+			})
+		}
+	}
+	return
+}
+
+// appendUniqueImpactPathsForMultipleRoots appends the source impact path to the target impact path while avoiding duplicates.
+// Specifically, it is designed for handling multiple root projects, such as Maven or Gradle, by comparing each pair of paths and identifying the path that is closest to the direct dependency.
+func AppendUniqueImpactPathsForMultipleRoots(target [][]services.ImpactPathNode, source [][]services.ImpactPathNode) [][]services.ImpactPathNode {
+	for targetPathIndex, targetPath := range target {
+		for sourcePathIndex, sourcePath := range source {
+			var subset []services.ImpactPathNode
+			if len(sourcePath) <= len(targetPath) {
+				subset = isImpactPathIsSubset(targetPath, sourcePath)
+				if len(subset) != 0 {
+					target[targetPathIndex] = subset
+				}
+			} else {
+				subset = isImpactPathIsSubset(sourcePath, targetPath)
```

Comment:
I understand what you are doing here when len(source path) > len(target path) but it doesnt match the description you gave the function. Maybe change the function description of isImpactPathIsSubset to:
"checks if first argument path is a subset of the second argument path" or something like this

Reply:
This content just moved. the PR did not change it. this is how we are calculating the Impact path.
It will change soon in the next PR's